hoomd/hpmc/update.py

```diff
@@ -472,6 +473,198 @@ def N(self):  # noqa: N802 - allow N as a function name
+            parameter = getattr(self._cpp_obj, attr)
+            return parameter
+        else:
+            return self._param_dict[attr]
+
+    def _setattr_param(self, attr, value):
+        if attr == "shape_move":
+            self._param_dict["shape_move"] = value
+        else:
+            super()._setattr_param(attr, value)
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be a HPMC integrator.")
+        if not integrator._attached:
+            raise RuntimeError("Integrator is not attached yet.")
+
+        updater_cls = None
+        shapes = [
+            'Sphere', 'ConvexPolygon', 'SimplePolygon', 'ConvexPolyhedron',
+            'ConvexSpheropolyhedron', 'Ellipsoid', 'ConvexSpheropolygon',
+            'Polyhedron', 'Sphinx', 'SphereUnion'
+        ]
+        for shape in shapes:
+            if isinstance(integrator, getattr(integrate, shape)):
+                updater_cls = getattr(_hpmc, 'UpdaterShape' + shape)
+        if updater_cls is None:
+            raise RuntimeError("Integrator not supported")
+        # TODO: Make this possible
+        # Currently computing the moments of inertia for spheropolyhedra is not implemented
+        # In order to prevent improper usage, we throw an error here. The use of this
+        # updater with spheropolyhedra is currently enabled to allow the use of spherical
+        # depletants
+        if isinstance(integrator, integrate.ConvexSpheropolyhedron):
+            for typ in integrator.type_shapes:
+                if typ['sweep_radius'] != 0 and len(typ['vertices']) > 1:
+                    raise RuntimeError(
+                        "Currently alchemical moves with integrate.convex_spheropolyhedron are only enabled for polyhedral and spherical particles."
+                    )
+        self._attach_shape_move(self._simulation)
+        self._cpp_obj = updater_cls(self._simulation.state._cpp_sys_def,
+                                    integrator._cpp_obj,
+                                    self.shape_move._cpp_obj, self.nselect,
+                                    self.nsweeps, self.pretend,
+                                    self.multi_phase, self.num_phase)
+        super()._attach()
+
+    @log(category='sequence')
+    def shape_moves(self):
+        """tuple[int, int]: Count of the accepted and rejected shape moves.
+
+        None when not attached
+        """
+        if self._attached:
+            total = self._cpp_obj.total_count
+            accepted = self._cpp_obj.accepted_count
+            return (accepted, total - accepted)
+        else:
+            return (0, 0)
+
+    @log(category='scalar')
+    def particle_volume(self):
+        """float: Total volume being occupied by particles.
+
+        None when not attached
+
+        TODO - move this to the shape moves classes
```

Comment:
Do we still plan on addressing this todo?

Reply:
It may be a bit more involved than I thought, so its fine to leave it in the `UpdaterShape`  class